docs/source/user/vectorize.rst

```diff
@@ -215,6 +215,33 @@ complicated inputs, depending on their shapes::
    Use it to ensure the generated code does not fallback to
    :term:`object mode`.
 
+
+.. _scalar-return-values:
+
+Scalar return values
+------------------------
+
+Now suppose we want to return a scalar value from :func:`~numba.guvectorize`::
+
+   @guvectorize([(int64[:], int64, int64[:])], '(n),()->()')
+   def g(x, y, res):
+       acc=0
+       for i in range(x.shape[0]):
+           acc += x[i] + y
+       res[0]=acc
+
+The modified function takes the sum of the 1-dimensional array (``x``) plus the scalar (``y``).
+There are a few important changes that allow the wrapped function to return a scalar:
+
+* in the signatures, declare the scalar return with ``[:]`` like a 1-dimensional array (eg. ``int64[:]``)
+* in the layout, declare it as ``()``
+* in the implementation, write the return value to output. (e.g. ``res[0]=acc``)
```

Comment:
I think it would be easier to digest if we write the explanation of how to return scalars then follow with an example. E.g.:

```suggestion
Now suppose we want to return a scalar value from :func:`~numba.guvectorize`. To do this, we need to:

* in the signatures, declare the scalar return with ``[:]`` like a 1-dimensional array (eg. ``int64[:]``),
* in the layout, declare it as ``()``,
* in the implementation, write to the first element (e.g. ``res[0] = acc``).

The following example function computes the sum of the 1-dimensional array (``x``) plus the scalar (``y``) and returns it as a scalar:
```

(then the example follows afterwards). Note there are some minor edits to each point in the suggestion above too.

Reply:
Thanks for the review. I'll add this example as a doc test and address the other samples in a follow-up PR.